src/state/slices/swapSlice/selectors.ts

```diff
@@ -0,0 +1,26 @@
+import type { Swap } from '@shapeshiftoss/swapper'
+import { createSelector } from 'reselect'
+
+import { swapSlice } from './swapSlice'
+
+import { selectQuoteIdParamFromFilter } from '@/state/selectors'
+
+export const selectSwaps = swapSlice.selectors.selectSwaps
+
+export const selectSwapById = createSelector(selectSwaps, swaps => {
+  return swaps.reduce(
+    (acc, swap) => {
+      acc[swap.id] = swap
+      return acc
+    },
+    {} as Record<string, Swap>,
+  )
+})
+
+export const selectSwapByQuoteId = createSelector(
+  selectSwaps,
+  selectQuoteIdParamFromFilter,
+  (swaps, quoteId) => {
+    return swaps.find(s => s.quoteId === quoteId)
```

Comment:
That looks off - isn't the whole intent of this to decouple swaps from quotes? 
Shouldn't we rather ensure that a *swap* gets upserted on preview, and then select by swapId?

Reply:
I have been able to decouple the swap by having a notion of currentSwap that we can retrieve everywhere in the flow

I had to move the swap upserting at confirm time instead of in the `useTradeActionSubscriber` hook, this permitted to remove all the flaky logic of the useEffect from it as the swap is now providing all the necessary data